WalletWasabi/Blockchain/Keys/KeyManager.cs

```diff
@@ -771,4 +681,27 @@ public void AssertNetworkOrClearBlockState(Network expectedNetwork)
 	}
 
 	#endregion BlockchainState
+
+	private static HdPubKey CreateHdPubKey((KeyPath KeyPath, ExtPubKey ExtPubKey) x) =>
+		new (x.ExtPubKey.PubKey, x.KeyPath, SmartLabel.Empty, KeyState.Clean);
+}
+
+public static class KeyPathExtensions
+{
+	public static ScriptPubKeyType GetScriptTypeFromKeyPath(this KeyPath keyPath) =>
+		keyPath.ToBytes().First() switch
+		{
+			84 => ScriptPubKeyType.Segwit,
+			86 => ScriptPubKeyType.TaprootBIP86,
+			_ => ScriptPubKeyType.Segwit // User can specify a specify whatever (like m/999'/999'/999')
+			// throw new NotSupportedException("Unknown script type.")
+		};
 }
+public static class HdPubKeyExtensions
+{
+	public static BitcoinAddress GetAddress(this HdPubKey me, Network network) =>
+		me.PubKey.GetAddress(me.FullKeyPath.GetScriptTypeFromKeyPath(), network);
+
+	public static Script GetAssumedScriptPubKey(this HdPubKey me) =>
```

Comment:
You are right I was not clear on this. Elaborate on this in some way because I don't understand what is this returning to me. It is blocking the review.